src/core/datastore/networkstore.js

```diff
@@ -24,6 +24,21 @@ export class NetworkStore {
      */
     this.collection = collection;
 
+    if (options.tag) {
+      let tag = options.tag;
+
+      if (!isString(tag)) {
+        throw new KinveyError('options.tag must be a string.');
+      }
+
+      tag = tag.trimRight();
+      if (tag === '') {
```

Comment:
When I said "not every symbol is valid", I meant we should do some research, which symbols are invalid database symbols for the different storage providers and add them to a list. Along with whitespace characters, and check with regex.

And from this, we can address my other comment about the separator between the collection name and the tag in the database collection name, which I feel holds about half the value of validating the tag in the first place. We want to choose a symbol, disallowed for a tag and collection character, but available for a collection name in the database.

I'm sorry if this is getting tedious. We can talk it over before doing the changes, as always. It's just my opinion.

Reply:
I can validate the tag against the regex we use for collection names. `^[a-zA-Z0-9-]*$`

Reply:
That regex would deem an empty string valid - it should be a plus instead of a star, at the end. So I guess "_" is a good candidate for our separateor

Reply:
I have added the regex `^[a-zA-Z0-9-]+$` for validation. I concatenated the collection name and tag together with a `.`